File: libpretixsync/src/main/java/eu/pretix/libpretixsync/sync/InvoicesettingsSyncAdapter.java
package eu.pretix.libpretixsync.sync;

import org.json.JSONException;
import org.json.JSONObject;

import eu.pretix.libpretixsync.api.PretixApi;
import eu.pretix.libpretixsync.db.Invoicesettings;
import io.requery.BlockingEntityStore;
import io.requery.Persistable;

public class InvoicesettingsSyncAdapter extends BaseSingleObjectSyncAdapter<Invoicesettings> {
    public InvoicesettingsSyncAdapter(BlockingEntityStore<Persistable> store, String eventSlug, String key, PretixApi api, SyncManager.ProgressFeedback feedback) {
        super(store, eventSlug, key, api, feedback);
    }

    @Override
    Invoicesettings getKnownObject() {
        return store.select(Invoicesettings.class)
                .where(Invoicesettings.SLUG.eq(key))
                .get().firstOrNull();
    }

    @Override
    public void updateObject(Invoicesettings obj, JSONObject jsonobj) throws JSONException {
        obj.setSlug(eventSlug);
        obj.setName(jsonobj.getString("invoice_address_from_name"));
        obj.setAddress(jsonobj.getString("invoice_address_from"));
        obj.setZipcode(jsonobj.getString("invoice_address_from_zipcode"));
        obj.setCity(jsonobj.getString("invoice_address_from_city"));
        obj.setCountry(jsonobj.getString("invoice_address_from_country"));
        obj.setTax_id(jsonobj.getString("invoice_address_from_tax_id"));
        obj.setVat_id(jsonobj.getString("invoice_address_from_vat_id"));
        obj.setJson_data(jsonobj.toString());
    }

    @Override
    protected String getUrl() {
        return api.eventResourceUrl("invoicesettings");
    }

    @Override
    String getResourceName() {
        return "invoicesettings";
    }

    @Override
    Invoicesettings newEmptyObject() {
        return new Invoicesettings();
    }
}

File: libpretixsync/src/main/java/eu/pretix/libpretixsync/sync/InvoiceSettingsSyncAdapter.kt
package eu.pretix.libpretixsync.sync

import eu.pretix.libpretixsync.api.PretixApi
import eu.pretix.libpretixsync.sqldelight.SyncDatabase
import eu.pretix.libpretixsync.sync.SyncManager.ProgressFeedback

class InvoiceSettingsSyncAdapter(
    db: SyncDatabase,
    eventSlug: String,
    key: String,
    api: PretixApi,
    syncCycleId: String,
    feedback: ProgressFeedback? = null,
) : SettingsSyncAdapter(
    db = db,
    eventSlug = eventSlug,
    key = key,
    api = api,
    syncCycleId = syncCycleId,
    feedback = feedback,
) {
    override fun getUrl(): String {
        return api.eventResourceUrl(eventSlug, "invoicesettings")
    }
}
